src/app/cityio/cityio.service.ts

```diff
@@ -0,0 +1,56 @@
+import { Injectable } from "@angular/core";
+import { HttpClient } from "@angular/common/http";
+import { Observable, of } from "rxjs";
+import { catchError, tap, retry } from "rxjs/operators";
+
+const CITYIO_TABLE_URL = "https://cityio.media.mit.edu/api/table/";
+const DEFAULT_TABLE_NAME = "grasbrook";
+
+@Injectable({
+  providedIn: "root"
+})
+export class CityioService {
+  cityIOTableURL: string;
```

Comment:
type URL?

Reply:
how can `"./assets/cityIO_demo.json"` be returned as Url type?